web-common/src/features/dashboards/filters/dimension-filters/DimensionFilter.svelte

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+  import {
+    defaultChipColors,
+    excludeChipColors,
+  } from "@rilldata/web-common/components/chip/chip-types";
+  import RemovableListChip from "../../../../components/chip/removable-list-chip/RemovableListChip.svelte";
+  import { getFilterSearchList } from "../../selectors/index";
+  import { getStateManagers } from "../../state-managers/state-managers";
+  import DimensionFilterReadOnlyChip from "./DimensionFilterReadOnlyChip.svelte";
+
+  export let name: string;
+  export let label: string;
+  export let column: string;
+  export let selectedValues: string[];
+  export let readOnly: boolean;
+
+  const StateManagers = getStateManagers();
+  const {
+    dashboardStore,
+    actions: {
+      dimensionsFilter: { toggleDimensionFilterMode },
+    },
+  } = StateManagers;
+
+  $: isInclude = !$dashboardStore.dimensionFilterExcludeMode.get(name);
+
+  let isOpen = false;
+  let searchText = "";
+  let allValues: Record<string, string[]> = {};
+  let topListQuery: ReturnType<typeof getFilterSearchList> | undefined;
+
+  $: if (isOpen) {
+    topListQuery = getFilterSearchList(StateManagers, {
```

Comment:
We should disable this query if it is readonly. Even though `isOpen` is not set to true right now we might end up changing it. Perhaps just a comment would be enough to make sure future devs will keep it in mind?

Reply:
Kk, I just added the conditional now.